src/jit/core_templates.expr

```diff
@@ -381,18 +498,122 @@
       (carg $2 ptr)
       (carg $3 int)) int_sz))
 
+(template: escape
+  (call (^func &MVM_string_escape)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 ptr)) ptr_sz))
+
 (template: flip
   (call (^func &MVM_string_flip)
     (arglist
       (carg (tc) ptr)
       (carg $1 ptr)) ptr_sz))
 
+(template: setbuffersize_fh!
+  (callv (^func &MVM_io_set_buffer_size)
+    (arglist
+      (carg (tc) ptr)
+      (carg $0 ptr)
+      (carg $1 int))))
+
+(template: iscclass
+  (call (^func &MVM_string_is_cclass)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 int)
+      (carg $2 ptr)
+      (carg $3 int)) int_sz))
+
+(template: findcclass
+  (call (^func &MVM_string_find_cclass)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 int)
+      (carg $2 ptr)
+      (carg $3 int)
+      (carg $4 int)) int_sz))
+
+(template: findnotcclass
+  (call (^func &MVM_string_find_not_cclass)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 int)
+      (carg $2 ptr)
+      (carg $3 int)
+      (carg $4 int)) int_sz))
+
+(template: nfafromstatelist
+  (call (^func &MVM_nfa_from_statelist)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 ptr)
+      (carg $2 ptr)) ptr_sz))
+
+(template: nfarunproto
+  (call (^func &MVM_nfa_run_proto)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 ptr)
+      (carg $2 ptr)
+      (carg $3 int)) ptr_sz))
+
+(template: nfarunalt!
+  (callv (^func &MVM_nfa_run_alt)
+    (arglist
+      (carg (tc) ptr)
+      (carg $0 ptr)
+      (carg $1 ptr)
+      (carg $2 int)
+      (carg $3 ptr)
+      (carg $4 ptr)
+      (carg $5 ptr))))
+
+(template: radix
+  (call (^func &MVM_radix)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 int)
+      (carg $2 ptr)
+      (carg $3 int)
+      (carg $4 int)) ptr_sz))
+
+(template: encode
+  (call (^func &MVM_string_encode_to_buf)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 ptr)
+      (carg $2 ptr)
+      (carg $3 ptr)
+      (carg (const 0 ptr_sz) ptr)) ptr_sz))
+
+(template: decode
+  (call (^func &MVM_string_decode_from_buf)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 ptr)
+      (carg $2 ptr)) ptr_sz))
+
+(template: istrue_s
+  (call (^func &MVM_coerce_istrue_s)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1 ptr)) int_sz))
+
+(template: isfalse_s
+  (if
+    (nz
+      (call (^func &MVM_coerce_istrue_s)
+        (arglist
+          (carg (tc) ptr)
+          (carg $1 ptr)) int_sz))
+    (const 0 int_sz)
+    (const 1 int_sz)))
+
 (template: null (^vmnull))
 
 (template: isnull
-  (or
-    (flagval (zr $1))
-    (flagval (eq $1 (^vmnull)))))
+  (if (^is_vmnull $1) (const 1 int_sz) (const 0 int_sz)))
```

Comment:
This will usually compile into bigger, branchier code (that maybe an optimizer can eliminate). But for clarity it is better.